google/cloud/documentai_toolbox/wrappers/document_revision.py

```diff
@@ -0,0 +1,346 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Wrappers for Document AI Document type."""
+
+import dataclasses
+import re
+import copy
+
+from typing import List, TypeVar
+
+from google.cloud import documentai
+from google.cloud import storage
+
+from google.cloud.documentai_toolbox.wrappers.entity import Entity
+
+from google.cloud.documentai_toolbox.wrappers.document import Document
+
+_OP_TYPE = documentai.Document.Provenance.OperationType
+
+DocumentWithRevisions = TypeVar("DocumentWithRevisions")
+
+
+def _get_base_and_revision_bytes(output_bucket: str, output_prefix: str) -> List[bytes]:
+    r"""Returns a list of shards as bytes.
+
+    If the filepaths are gs://abc/def/gh/{1,2,3}.json, then output_bucket should be "abc",
+    and output_prefix should be "def/gh".
+
+    Args:
+        output_bucket (str):
+            Required. The name of the output_bucket.
+
+        output_prefix (str):
+            Required. The prefix of the folder where files are excluding the bucket name.
+
+    Returns:
+        List[bytes]:
+            A list of shards as bytes.
+
+    """
+    text = ""
+    base_doc = []
+    revisions_doc = []
+
+    storage_client = storage.Client()
+
+    blob_list = storage_client.list_blobs(output_bucket, prefix=output_prefix)
+    pb = documentai.Document.pb()
+    for blob in blob_list:
+        name = blob.name.split("/")[-1]
+
+        if re.search(r"^doc.dp.bp", name) != None:
+            if blob.name.endswith(".dp.bp"):
+                blob_as_bytes = blob.download_as_bytes()
+                text = pb.FromString(blob_as_bytes).text
+        if re.search(r"^pages_*.*", name) != None:
+            if blob.name.endswith(".dp.bp"):
+                blob_as_bytes = blob.download_as_bytes()
+                base_doc.append(blob_as_bytes)
+        elif re.search(r"^rev_*.*", name) != None:
+            if blob.name.endswith(".dp.bp"):
+                blob_as_bytes = blob.download_as_bytes()
+                revisions_doc.append(blob_as_bytes)
```

Comment:
I think this could greatly simplify the conditions here. Probably even more that could be done, but I'm not familiar with the exact name checks going on here.
```suggestion
        name = blob.name.split("/")[-1]
        if blob.name.endswith(".dp.bp"):
            blob_as_bytes = blob.download_as_bytes()
            if re.search(r"^doc.dp.bp", name):
                text = pb.FromString(blob_as_bytes).text
            elif re.search(r"^pages_*.*", name):
                base_doc.append(blob_as_bytes)
            elif re.search(r"^rev_*.*", name):
                revisions_doc.append(blob_as_bytes)
```